### ISSUE TITLE
🚀[FEA]: Address incorrect virial in `ewald_reciprocal_space()` with detached cells

### ISSUE THREAD

**Issue by sklenard** [NVIDIA/nvalchemi-toolkit-ops#147]
### Is this a new feature, an improvement, or a change to existing functionality?

Improvement

### How would you describe the priority of this feature request

Low (would be nice)

### Please provide a clear description of problem you would like to solve.

As pointed out in issue #136, taking a strain/cell gradient through `ewald_reciprocal_space()` produces an incorrect virial when using precomputed `k_vectors` (i.e. detached cell). In nvalchemiops 0.4.1 a `UserWarning` is now raised (see PR #138). 

An alternative solution to prevent this would be to modify `generate_k_vectors_ewald_summation()` to return `miller_indices` instead of `k_vectors` (since this is what is computed internally). Then `ewald_reciprocal_space()` could compute `k_vectors` internally as:
```python
reciprocal_cell = (
    TWOPI * torch.linalg.inv_ex(cell.transpose(1, 2))[0]
)  # Transpose for column vectors
k_vectors = miller_indices @ reciprocal_cell
```

### Describe any alternatives you have considered

This can be handled manually with the following workaround:
```python
# precompute miller indices
k_vectors = generate_k_vectors_ewald_summation(cell=cell, k_cutoff=k_cutoff)
k_miller = torch.round(
    torch.bmm(k_vectors, cell.detach().transpose(-1, -2)) / TWOPI 
)
...
# rebuild from the cell so ∂k/∂cell exists to get correct virial
k_vectors = TWOPI * torch.matmul(
    k_miller.to(cell.dtype),
    torch.linalg.inv(cell).transpose(-1, -2),
)
```
